### PR TITLE
PDPDEVTOOL-3952: [VSCode] Pass VSCode specific info to the core (SDK)…

### PR DESCRIPTION
**Pull request by karl-anthony-ng**
… for metrics collection - Platform, Platform Version

### CODE REVIEW

File: packages/node-cli/src/SdkExecutor.js [hunk 1]
```diff
@@ -29,7 +30,6 @@ module.exports = class SdkExecutor {
 		this._sdkPath = sdkPath;
 
 		this._CLISettingsService = new CLISettingsService();
-		this._environmentInformationService = new EnvironmentInformationService();
```
Comment (davidecorreu) [inline packages/node-cli/src/SdkExecutor.js:32]: The `this._environmentInformationService = new EnvironmentInformationService();` will still be needed by `_checkIfJavaVersionIssue()`

**Review by lfpvillegas — APPROVED** [review]

File: packages/node-cli/src/SdkExecutor.js (hunk 1 above)
Comment (karl-anthony-ng) [reply packages/node-cli/src/SdkExecutor.js:32]: Added back.

**Review by lfpvillegas — APPROVED** [review]

File: packages/node-cli/src/commands/Command.js [hunk 2]
```diff
@@ -38,9 +38,10 @@ class Command {
 		this._runInInteractiveMode = options.runInInteractiveMode;
 		this._interactiveSupport = options.interactiveSupport;
 		this._log = options.log;
+		this._executionEnvironmentContext = options.executionEnvironmentContext;
 
-		this._action = new action(options);
-		this._inputHandler = new inputHandler(options);
+		this._action = new action(options, this._executionEnvironmentContext);
+		this._inputHandler = new inputHandler(options, this._executionEnvironmentContext);
```
Comment (davidecorreu) [inline packages/node-cli/src/commands/Command.js:44]: First of all, great refactor Karl it looks much cleaner than the previous implementation.!!!
The executionEnvironmentContext is included in the options object, and we don't need to add an extra parameter when constructing the Action or InputHandler.
Actually in the BaseAction this second parameter is not used and the executionEnvironmentContet is take from the options. Same comment is valid for InputHandler. I also think that sdkPath can be added as a property in the BaseInputHandler
Snnipet from BaseAction:
```
module.exports = class BaseAction {
	constructor(options) {
		this._projectFolder = options.projectFolder;
		this._commandMetadata = options.commandMetadata;
		this._executionPath = options.executionPath;
		this._runInInteractiveMode = options.runInInteractiveMode;
		this._log = options.log;
		this._sdkPath = options.sdkPath;


		this._sdkExecutor = new SdkExecutor(this._sdkPath);
		this._executionEnvironmentContext = options.executionEnvironmentContext;
		this._sdkExecutor = new SdkExecutor(this._sdkPath, this._executionEnvironmentContext);
	}
```

File: packages/node-cli/src/commands/file/import/ImportFilesInputHandler.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ module.exports = class ImportFilesInputHandler extends BaseInputHandler {
 	constructor(options) {
 		super(options);
 		// TODO input handlers shouldn't execute actions. rework this
-		this._sdkExecutor = new SdkExecutor(options.sdkPath);
+		this._sdkExecutor = new SdkExecutor(options.sdkPath, this._executionEnvironmentContext);
```
Comment (davidecorreu) [inline packages/node-cli/src/commands/file/import/ImportFilesInputHandler.js:56]: If we add the sdkPath as a property of the BaseInputHandler we can rewrite this line as:
`this._sdkExecutor = new SdkExecutor(this._sdkPath, this._executionEnvironmentContext);` .

File: packages/node-cli/src/commands/Command.js (hunk 2 above)
Comment (karl-anthony-ng) [reply packages/node-cli/src/commands/Command.js]: I agree, it's looks much cleaner now.
Did some more updates based on your suggestion.

File: packages/node-cli/src/commands/file/import/ImportFilesInputHandler.js (hunk 3 above)
Comment (karl-anthony-ng) [reply packages/node-cli/src/commands/file/import/ImportFilesInputHandler.js]: Applied this change.

**Review by davidecorreu — APPROVED** [review]

**Pull request merged**